pkg/solana/txm/utils.go

```diff
@@ -69,3 +71,35 @@ func convertStatus(res *rpc.SignatureStatusesResult) uint {
 
 	return NotFound
 }
+
+type signatureList struct {
+	sigs []solana.Signature
+	lock sync.RWMutex
+}
+
+func (s *signatureList) Get(index int) (sig solana.Signature, err error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	if index >= len(s.sigs) {
+		return sig, errors.New("invalid index")
+	}
+	return s.sigs[index], nil
+}
+
+func (s *signatureList) List() []solana.Signature {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	return s.sigs
+}
+
+func (s *signatureList) Length() int {
+	return len(s.List())
```

Comment:
I think this is OK, but I had to read it carefully and know how slices work to understand that. What about this?
```suggestion
	s.lock.RLock()
	defer s.lock.RUnlock()
	return len(s.sigs)
```
Should be a little more efficient too.